cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -669,6 +674,30 @@ func (csr *ClusterStateRegistry) GetUnregisteredNodes() []UnregisteredNode {
 	return result
 }
 
+func (csr *ClusterStateRegistry) updateCloudProviderDeletedNodes(deletedNodes []*apiv1.Node) {
+	result := make(map[string]*apiv1.Node)
```

Comment:
nit: maybe set map capacity to `len(deletedNodes)+len(csr.deletedNodes)` to avoid multiple allocs?

Reply:
Agreed, I will include this in my next changes.